### DISCUSSION TITLE
是否可以优化下required的属性，当设置为false时，rules校验规则不对空值校验

### DISCUSSION THREAD

**Discussion by gary123123** [alibaba/formily#2442] [Q&A 1.x] [answered]
对于业务来说，当formIten的required设置为false时，说明当前输入框应该是允许置空的，所以校验规则不应该对空值进行校验，这样使用起来更为方便。
而antd就是这样的设计：https://stackblitz.com/edit/react-tgdw92
而formily则对required为false时，也对空值进行了校验：https://stackblitz.com/edit/react-uq7cz8

**Comment by janryWang** [comment, marked as answer]
不对，1.x是这样的，2.x其实已经做了空值过滤 https://github.com/alibaba/formily/blob/formily_next/packages/validator/src/rules.ts
